profiles/views.py

```diff
@@ -102,3 +102,26 @@ def destroy(self, request, pk=None):
         profile.save()
         serializer = ProfileSerializer(profile)
         return Response(serializer.data, status=status.HTTP_204_NO_CONTENT)
+
+
+class ViewedCompanyList(ListCreateAPIView):
+    queryset = ViewedCompany.objects.all()
+    serializer_class = ViewedCompanySerializer
+    permission_classes = (IsAuthenticated, )
+
+    def list(self, request, *args, **kwargs):
+        user_id = self.request.user.id
+        viewed_companies = ViewedCompany.objects.filter(user=user_id)
+        serializer = self.serializer_class(viewed_companies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def create(self, request, *args, **kwargs):
+        user_id = self.request.user.id
+        company_id = request.data.get("profile_id")
```

Comment:
Don't do manual manipulations with payload, unless there is no other way to achieve something.

This is work of serializer.

Reply:
Moved validation to serializer, removed overriding of `create` in the view.